### PR TITLE
[Torch] support decompose aten.einsum with ellipsis slicing

### PR DESCRIPTION
**Pull request by Xinyu302**
as title

### CODE REVIEW

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 1]
```diff
@@ -158,6 +159,99 @@ static SmallVector<int64_t> computeDimsOrderForMoveDim(int64_t srcDimInt,
...
+    int num = maxEllipsisRank;
+    char lastChar = 'a';
+    while (num--) {
+      while (usedTokens.find(lastChar) != usedTokens.end()) {
+        lastChar++;
+      }
+      token.push_back(lastChar);
+      lastChar++;
+    }
+    return token;
```
Comment (ramiro050) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:227]: nit: this does not need to be a lambda expression. I think `token` can just be replaced with `ellipsisToken` and the lambda expression can be removed.

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 2]
```diff
@@ -158,6 +159,99 @@ static SmallVector<int64_t> computeDimsOrderForMoveDim(int64_t srcDimInt,
...
+  // tensor can be broadcasted
+  SmallVector<int64_t> ellipsisRanks;
+  int maxEllipsisRank = 0;
+  for (const auto &[token, inputRank] : llvm::zip(inputTokens, inputRanks)) {
+    int explictRank = 0;
+    for (auto c : token) {
+      if (std::isalpha(c)) {
+        usedTokens.insert(c);
+        explictRank++;
+      } else if (c == '.' || c == ' ') {
```
Comment (ramiro050) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:201]: are blank spaces ever encountered in the equations?

**Review by ramiro050 — APPROVED** [review]
Just a couple of small comments, but other than that LGTM!

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp (hunk 2 above)
Comment (Xinyu302) [reply lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:201]: Actually, space is a legal input in einsum equations

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp (hunk 1 above)
Comment (Xinyu302) [reply lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp]: I have already made the modifications according to your suggestions.

**Review by ramiro050 — APPROVED** [review]
Thanks!

**Pull request merged**